Sources/Filters/General/ImageDataOutlineFilter/index.js

```diff
@@ -0,0 +1,149 @@
+      const p = points.subarray(i, i + 3);
+      if (transform.length === 9) {
+        vec3.transformMat3(tmp, p, transform);
+      } else {
+        vec3.transformMat4(tmp, p, transform);
+      }
+      [p[0], p[1], p[2]] = tmp;
+    }
+  }
+
+  publicAPI.requestData = (inData, outData) => {
+    // implement requestData
+    const input = inData[0];
+
+    if (!input || !input.isA('vtkImageData')) {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+
+    if (!model.tmpOut) {
+      // allocate output object if not done previously.
+      model.tmpOut = vtkPolyData.newInstance();
+    }
+
+    // First create a cube polydata in the index-space of the image.
+    const spatialExt = input.getSpatialExtent();
+    if (!spatialExt) {
+      vtkErrorMacro('Unable to fetch spatial extents of input image.');
+      return;
+    }
+
+    model.cubeSource.setXLength(spatialExt[1] - spatialExt[0]);
+    model.cubeSource.setYLength(spatialExt[3] - spatialExt[2]);
+    model.cubeSource.setZLength(spatialExt[5] - spatialExt[4]);
+
+    model.cubeSource.setCenter(
+      model.cubeSource.getXLength() / 2.0,
+      model.cubeSource.getYLength() / 2.0,
+      model.cubeSource.getZLength() / 2.0
+    );
+
+    model.cubeSource.update();
+
+    const out = model.cubeSource.getOutputData();
+    model.tmpOut.getPoints().deepCopy(out.getPoints());
+
+    // Now, transform the cube polydata points in-place
+    // using the image's indexToWorld transformation.
+    transformPoints(
+      model.tmpOut.getPoints().getData(),
+      input.getIndexToWorld()
+    );
+
+    // Lastly, generate the necessary cell arrays.
+    if (model.generateFaces) {
+      model.tmpOut.getPolys().deepCopy(out.getPolys());
+    } else {
+      model.tmpOut.getPolys().initialize();
+    }
+    if (model.generateLines) {
+      model.tmpOut.getLines().deepCopy(model.lineCells);
+    } else {
+      model.tmpOut.getLines().initialize();
+    }
+
+    model.tmpOut.modified();
+    outData[0] = model.tmpOut;
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  generateFaces: false,
+  generateLines: true,
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Make this a VTK object
+  macro.obj(publicAPI, model);
+
+  macro.setGet(publicAPI, model, ['generateFaces', 'generateLines']);
+
+  // Also make it an algorithm with one input and one output
+  macro.algo(publicAPI, model, 1, 1);
+
+  // Internal persistent objects
+  model.cubeSource = vtkCubeSource.newInstance();
+  model.tmpOut = vtkPolyData.newInstance();
+  model.lineCells = vtkCellArray.newInstance({
+    values: Uint16Array.from(LINE_ARRAY),
+  });
```

Comment:
should be made protected

Reply:
Updated. Please re-review to ensure if it was done correctly.

Reply:
looks good to me.